src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressHelper.cs

```diff
@@ -113,9 +128,39 @@ private static async Task GetStream(Stream outputStream, CancellationToken cance
         {
             const int DefaultBufferSize = 0x10000;
 
-            StdInStream = Console.OpenStandardInput();
             await StdInStream.CopyToAsync(outputStream, DefaultBufferSize, cancellationToken);
         }
+
+        // TODO: For .NET 7 and above, can use Stream.ReadExactly
+        private static async Task<int> ReadHelper(Memory<byte> buffer, long bufferSize, CancellationToken token)
```

Comment:
Let's try this:
- Make this method `Task ReadExactlyAsync(Memory<byte> buffer, CancellationToken)`; read exactly the number of bytes from the stream as there is room for in the buffer. Throw `EndOfStreamExecption` if cannot fill the buffer. Use `#if NET7_0_OR_GREATER` to defer to the `Stream.ReadExactlyAsync` method and fallback to your implementation for .NET 6 and lower.
- Remove the `ReadFromStream` method and have the former callers (1) `stackalloc` their `byte[]` and (2) pass it into `ReadExactlyAsync`.